src/dictionary/manager.ts

```diff
@@ -294,19 +294,23 @@ class SchemaManager {
   };
 }
 
-export const revalidateAllDonorClinicalEntitiesAgainstSchema = (
+export const revalidateAllDonorClinicalEntitiesAgainstSchema = async (
   donor: DeepReadonly<Donor>,
   schema: dictionaryEntities.SchemasDictionary,
 ) => {
   const clinicalSchemaNames = getSchemaNamesForDonorClinicalEntities(donor);
   let isValid = true;
-  clinicalSchemaNames.forEach((schemaName: ClinicalEntitySchemaNames) => {
+  for (const schemaName of clinicalSchemaNames) {
```

Comment:
Async only works properly in new `For of` loops

Reply:
unclear where the async function is. Was this never working?

Reply:
Let me work back through this; dictionaryService.processRecords is the root of the issue and the fact that we're sometimes revalidating hundreds of records.

Reply:
But, may have got my wires crossed

Reply:
OK, I was able to de-Promisify across several files:
https://github.com/icgc-argo/argo-clinical/pull/1051/commits/5ee92de3cf23fd50c26925e17c9db86183550279

Debugging is a windy road. GQL was returning null values, but getClinicalErrors was returning correct values, so it appeared race condition was a possibility.

Reply:
Much improved this way